File: eigenchannel correlation/trans_different_mode_process_data.py
# -*- coding: utf-8 -*-
"""
Created on Tue Feb  6 15:21:12 2018

@author: user
"""
# correlated pattern
import numpy as np
import matplotlib.pyplot as plt
#import random
#import kwant
from kwant.digest import uniform
#import time
import scipy.io as sio
import os.path

width=50
length=100
shape_whole=[]
shape_tau1=[]
v_dist=[]
t_list=[]
tau_list=[]
#final_tau_sqrt=[]
save_path = 'C:/Users/ykang/Documents/correlation eigenchannel/2/'
for cishu in np.arange(0,1000):#range(1000):
    completeName = os.path.join(save_path, str(cishu)+".mat")
#    myDict = {'u':u,'v':v,'s':s,'wf':wf,'T':T,'gf_mode':gf_mode}

    wf=sio.loadmat(completeName)['wf']
    wf_ave=np.mean(abs(wf)**2,axis=0)
    wf_ave_reshape=wf_ave.reshape((length,width))
    wf_shape=np.sum(wf_ave_reshape,axis=1)
    shape_whole.append(wf_shape)
    s=sio.loadmat(completeName)['s']
    tau_list.append(s**2)
    if s[0,0]>0.98:
        v=sio.loadmat(completeName)['v']
        f_tau1=abs(np.array(sum(np.dot(np.matrix(np.diag(v[0,:])).H,wf))))**2
        f_tau1_reshape=f_tau1.reshape((length,width))
        shape_tau1.append(np.sum(f_tau1_reshape,axis=1))
#        shape.append(shape_tau1)
#        plt.figure()
#        plt.imshow(ff_tau1)
#        v_dist.append(abs(v[0,:])**2)
        
    t=sio.loadmat(completeName)['T']
    t_list.append(t)
    
plt.title('ave_whole eigenchannel')
plt.plot(np.mean(np.array(shape_whole),axis=0))
plt.figure()
plt.plot(np.mean(np.array(shape_tau1),axis=0))
plt.title('ave_tau1')
#plt.figure()
#v_mean=np.mean(np.array(v_dist),axis=0)
#plt.plot(v_mean)
#plt.title('ave_v_tau1')
print(np.mean(t_list))

#%%
#plt.imshow(pattern)
#plt.imshow(f_tau1_reshape)
plt.figure()
plt.hist(np.array(tau_list).reshape((-1,1)))

File: src/others/trans_different_mode_process_data.m
clear
width=49;
length=159;
shape_whole=[];
shape_tau1=[];
v_dist=[];
k=1;
m=1;
save_path = 'E:/correlation pattern/10/';
% load([save_path,'velocity.mat'])
for cishu =0:499
    load([save_path,num2str(cishu),'.mat'])
    wf_ave=mean(abs(wf).^2,1);  %diag(1./mode_v)*abs(wf).^2
    wf_ave_reshape=reshape(wf_ave,width,length);
    wf_shape=sum(wf_ave_reshape,1);
    shape_whole(k,:)=wf_shape;
   
    

    
%     if s(1)>0.98
%         f_tau1=abs(sum(diag(v(1,:))'*wf,1)).^2;
%         f_tau1_reshape=reshape(f_tau1,width,length);
%         shape_tau1(m,:)=sum(f_tau1_reshape,1);      
% 
%         v_dist(m,:)=abs(v(1,:)).^2;
%         m=m+1;
%     end
    k=k+1;
end
% x=linspace(0,1,length);
figure(6)
hold on
% title('ave whole eigenchannel')
l=1:length;
temp=mean(shape_whole,1);
plot(l,temp)

c = polyfit(l,temp,1);
l0=-c(2)/c(1);

y_est = polyval(c,[l0 l]);
hold on
plot([l0 l],y_est,'r--')
xlabel('sample length')
ylabel('<W(x)>')

legend('n=1 zb=18.45','n=1.1 zb=19.16','n=1.2 zb=22.63','n=1.5 zb=34.42')
% figure(2)
% hold on
% x=(0:length-1)/(length-1);
% plot(x,mean(shape_tau1,1),'-')
% legend('l=159','l=99','l=39')
% ylabel('average of tau=1 eigenchannel')
% xlabel('x/l')
% title('ave tau1')
% figure(3)
% hold on
% v_mean=mean(v_dist,1);
% plot(v_mean)
% title('ave v tau1')
% figure
% % 1/mean(1./mode_v)
% wf_1_reshape=reshape(wf(4,:),width,length);
% wf1_shape=sum(wf_1_reshape,1);
% plot(abs(wf1_shape).^2)

% zb=[];
l0-length